### PR TITLE
llvm-project@17.0.3.bcr.1 - Add patch for incompatible_disallow_empty_glob

### PR DESCRIPTION
**Pull request by UebelAndre**
An issue was reported in `rules_rust` (https://github.com/bazelbuild/rules_rust/issues/3137) where some local patches to satisfy https://github.com/bazelbuild/bazel/issues/8195 were added to llvm via a `single_version_override` which does not affect consumers. This change is an attempt to upstream an otherwise noop change to llvm that supports this flag being flipped in Bazel.

### CODE REVIEW

**Comment by bazel-io** [comment]
Hello @bazelbuild/bcr-maintainers, modules without existing maintainers (llvm-project) have been updated in this PR. Please review the changes.

**Review by fmeum — COMMENTED** [review]
Could you replace the otherwise unmodified patches with relative symlinks to the previous version?

**Comment by UebelAndre** [comment]
@fmeum updated!

File: modules/llvm-project/17.0.3.bcr.1/presubmit.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+matrix:
+  platform:
+  - debian10
+  - ubuntu2004
+  bazel:
+  - "7.x"
```
Comment (fmeum) [inline modules/llvm-project/17.0.3.bcr.1/presubmit.yml:6]:
```suggestion
  - "7.x"
  - "8.x"
```

**Review by fmeum — APPROVED** [review]

File: modules/llvm-project/17.0.3.bcr.1/patches/0003-Add-MODULE.bazel.patch [hunk 2]
```diff
@@ -0,0 +1 @@
+../../17.0.3/patches/0003-Add-MODULE.bazel.patch
```
Comment (fmeum) [inline modules/llvm-project/17.0.3.bcr.1/patches/0003-Add-MODULE.bazel.patch:1]: This patch needs to be modified to use the new version

File: modules/llvm-project/17.0.3.bcr.1/presubmit.yml (hunk 1 above)
Comment (UebelAndre) [reply modules/llvm-project/17.0.3.bcr.1/presubmit.yml:6]: I'll add this but I strongly feel Bazel 8 support should not be required for this change if it doesn't immediately work.

File: modules/llvm-project/17.0.3.bcr.1/patches/0003-Add-MODULE.bazel.patch (hunk 2 above)
Comment (UebelAndre) [reply modules/llvm-project/17.0.3.bcr.1/patches/0003-Add-MODULE.bazel.patch]: oops!

File: modules/llvm-project/17.0.3.bcr.1/presubmit.yml (hunk 1 above)
Comment (fmeum) [reply modules/llvm-project/17.0.3.bcr.1/presubmit.yml:6]: It never is, but we should always give it a try :⁠-⁠)

File: modules/llvm-project/17.0.3.bcr.1/presubmit.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  - "8.x"
```
Comment (fmeum) [inline modules/llvm-project/17.0.3.bcr.1/presubmit.yml:7]:
```suggestion
```

**Review by fmeum — APPROVED** [review]
Bazel 8 would have passed if I hadn't introduced a bug that will only be fixed in 8.0.1.

**Pull request merged**